contracts/Auction.sol

```diff
@@ -113,7 +113,12 @@ contract Auction {
 
 
     function endAuction() public {
+        require(now >=  end_time; "Auction not yet ended.");
+        require(!ended, "auction  has already ended.");
 
+ 
+        ended = true;
+        bidder_address.transfer(highest_bid);
```

Comment:
do we have `highest_bid ` member ?

Reply:
Yes, they are. You can also check. one is in the struct. 
and another is defined also.